### DISCUSSION TITLE
Timeline for instance segmentation + training

### DISCUSSION THREAD

**Discussion by michaellee1** [open-mmlab/mmyolo#730] [Q&A]
We're looking to build on top of mmyolo to support training and running YOLO-based instance segmentation models.  I was hoping to get an updated timeline of when this might be available to decide whether we should build on top of mmyolo or on top of the yolov8 repository.  We'd ideally love to build on top of this, as it fits in well with the rest of the mmdetection libraries.

I know that the team has mentioned 3 weeks ago that you are currently building this, in a Github issue I looked at.  On an older issue I saw a contributor mentioned it potentially being released in March.  Do you have an estimate of when instance segmentation might be ready to train and run using this repo?  Is it within weeks, or months, or more?

I also noticed that it was mentioned that inference would be supported first, followed by training.  Will training be significantly more delayed than inference?

Thank you so much!

**Comment by hhaAndroid** [comment]
@michaellee1 It will definitely be released in April.

**Comment by michaellee1** [comment]
that's super helpful! @hhaAndroid just verifying, this will be both the inference and training code released together?

**Comment by michaellee1** [comment]
@hhaAndroid Any updates here?

**Comment by JKelle** [comment]
Hi @hhaAndroid, I am also interested in instance segmentation training. I see there is already a `RTMDetInsSepBNHead` class but it currently fails to train because `loss_by_feat` hasn't been updated to accept mask losses. 

Do you have an updated timeline? 

Thanks a lot!
